File: epi_judge_java/epi/SwapBits.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class SwapBits {
  @EpiTest(testDataFile = "swap_bits.tsv")
  public static long swapBits(long x, int i, int j) {
    long iBit = (x >>> i) & 1;
    long jBit = (x >>> j) & 1;
    if (iBit == jBit)
      return x;

    long mask = (1L << i) | (1L << j);
    return x ^ mask;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "SwapBits.java",
                new Object() {
                }.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_python/swap_bits.py
from test_framework import generic_test


def swap_bits(x, i, j):
    iBit = (x >> i) & 1
    jBit = (x >> j) & 1
    if iBit == jBit:
        return x

    mask = (1 << i) | (1 << j)
    return x ^ mask


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('swap_bits.py', 'swap_bits.tsv',
                                       swap_bits))
